[2024-11-28T20:47:07.030+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2024-11-28T20:47:07.041+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: brewery_ETL.transform_gold manual__2024-11-28T20:29:14.545680+00:00 [queued]>
[2024-11-28T20:47:07.047+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: brewery_ETL.transform_gold manual__2024-11-28T20:29:14.545680+00:00 [queued]>
[2024-11-28T20:47:07.047+0000] {taskinstance.py:2866} INFO - Starting attempt 2 of 2
[2024-11-28T20:47:07.056+0000] {taskinstance.py:2889} INFO - Executing <Task(PythonOperator): transform_gold> on 2024-11-28 20:29:14.545680+00:00
[2024-11-28T20:47:07.059+0000] {standard_task_runner.py:72} INFO - Started process 269 to run task
[2024-11-28T20:47:07.062+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'brewery_ETL', 'transform_gold', 'manual__2024-11-28T20:29:14.545680+00:00', '--job-id', '181', '--raw', '--subdir', 'DAGS_FOLDER/openbrewery_dag.py', '--cfg-path', '/tmp/tmpblwptkyw']
[2024-11-28T20:47:07.064+0000] {standard_task_runner.py:105} INFO - Job 181: Subtask transform_gold
[2024-11-28T20:47:07.101+0000] {task_command.py:467} INFO - Running <TaskInstance: brewery_ETL.transform_gold manual__2024-11-28T20:29:14.545680+00:00 [running]> on host 121ed6c9dece
[2024-11-28T20:47:07.162+0000] {taskinstance.py:3132} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='brewery_ETL' AIRFLOW_CTX_TASK_ID='transform_gold' AIRFLOW_CTX_EXECUTION_DATE='2024-11-28T20:29:14.545680+00:00' AIRFLOW_CTX_TRY_NUMBER='2' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-11-28T20:29:14.545680+00:00'
[2024-11-28T20:47:07.163+0000] {taskinstance.py:731} INFO - ::endgroup::
[2024-11-28T20:47:07.175+0000] {transform_gold.py:89} INFO - Iniciando a tarefa transform_gold
[2024-11-28T20:47:07.175+0000] {transform_gold.py:103} INFO - Data de execução: 2024-11-28
[2024-11-28T20:47:07.175+0000] {transform_gold.py:15} INFO - Iniciando criação da sessão Spark para transformação Gold
[2024-11-28T20:47:12.862+0000] {transform_gold.py:39} INFO - Sessão Spark criada com sucesso para transformação Gold
[2024-11-28T20:47:12.863+0000] {transform_gold.py:48} INFO - Lendo dados da camada Silver: s3a://breweris-etl-prod-sp/silver/date=2024-11-28/
[2024-11-28T20:47:15.182+0000] {transform_gold.py:56} ERROR - Erro ao ler dados da camada Silver: [UNABLE_TO_INFER_SCHEMA] Unable to infer schema for Parquet. It must be specified manually.
[2024-11-28T20:47:15.185+0000] {transform_gold.py:130} ERROR - Erro na tarefa transform_gold: [UNABLE_TO_INFER_SCHEMA] Unable to infer schema for Parquet. It must be specified manually.
[2024-11-28T20:47:15.757+0000] {transform_gold.py:135} INFO - Sessão Spark finalizada para a tarefa transform_gold
[2024-11-28T20:47:15.758+0000] {taskinstance.py:3311} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/models/baseoperator.py", line 417, in wrapper
    return func(self, *args, **kwargs)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/operators/python.py", line 238, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/operators/python.py", line 256, in execute_callable
    return runner.run(*self.op_args, **self.op_kwargs)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
  File "/opt/airflow/scripts/transform_gold.py", line 109, in transform_gold
    df_silver = read_silver_data(spark, s3_bucket, s3_silver_path, execution_date)
  File "/opt/airflow/scripts/transform_gold.py", line 50, in read_silver_data
    df_silver = spark.read.option("basePath", f"s3a://{s3_bucket}/{s3_silver_path}/") \
  File "/home/airflow/.local/lib/python3.9/site-packages/pyspark/sql/readwriter.py", line 531, in parquet
    return self._df(self._jreader.parquet(_to_seq(self._spark._sc, paths)))
  File "/home/airflow/.local/lib/python3.9/site-packages/py4j/java_gateway.py", line 1322, in __call__
    return_value = get_return_value(
  File "/home/airflow/.local/lib/python3.9/site-packages/pyspark/errors/exceptions/captured.py", line 175, in deco
    raise converted from None
pyspark.errors.exceptions.captured.AnalysisException: [UNABLE_TO_INFER_SCHEMA] Unable to infer schema for Parquet. It must be specified manually.
[2024-11-28T20:47:15.772+0000] {taskinstance.py:1225} INFO - Marking task as FAILED. dag_id=brewery_ETL, task_id=transform_gold, run_id=manual__2024-11-28T20:29:14.545680+00:00, execution_date=20241128T202914, start_date=20241128T204707, end_date=20241128T204715
[2024-11-28T20:47:15.787+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2024-11-28T20:47:15.787+0000] {standard_task_runner.py:124} ERROR - Failed to execute job 181 for task transform_gold ([UNABLE_TO_INFER_SCHEMA] Unable to infer schema for Parquet. It must be specified manually.; 269)
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/utils/cli.py", line 116, in wrapper
    return f(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/models/taskinstance.py", line 3005, in _run_raw_task
    return _run_raw_task(
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/models/taskinstance.py", line 273, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/models/taskinstance.py", line 3159, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/models/taskinstance.py", line 3183, in _execute_task
    return _execute_task(self, context, task_orig)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/models/baseoperator.py", line 417, in wrapper
    return func(self, *args, **kwargs)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/operators/python.py", line 238, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/operators/python.py", line 256, in execute_callable
    return runner.run(*self.op_args, **self.op_kwargs)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
  File "/opt/airflow/scripts/transform_gold.py", line 109, in transform_gold
    df_silver = read_silver_data(spark, s3_bucket, s3_silver_path, execution_date)
  File "/opt/airflow/scripts/transform_gold.py", line 50, in read_silver_data
    df_silver = spark.read.option("basePath", f"s3a://{s3_bucket}/{s3_silver_path}/") \
  File "/home/airflow/.local/lib/python3.9/site-packages/pyspark/sql/readwriter.py", line 531, in parquet
    return self._df(self._jreader.parquet(_to_seq(self._spark._sc, paths)))
  File "/home/airflow/.local/lib/python3.9/site-packages/py4j/java_gateway.py", line 1322, in __call__
    return_value = get_return_value(
  File "/home/airflow/.local/lib/python3.9/site-packages/pyspark/errors/exceptions/captured.py", line 175, in deco
    raise converted from None
pyspark.errors.exceptions.captured.AnalysisException: [UNABLE_TO_INFER_SCHEMA] Unable to infer schema for Parquet. It must be specified manually.
[2024-11-28T20:47:15.804+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 1
[2024-11-28T20:47:15.822+0000] {taskinstance.py:3895} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-11-28T20:47:15.824+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
